app/routers/event.py

```diff
@@ -25,16 +27,66 @@ async def eventedit(request: Request):
                                       {"request": request})
 
 
-@router.get("/view/{id}")
-async def eventview(request: Request, id: int):
+@router.get("/view/{event_id}")
+async def eventview(request: Request, event_id: int,
+                    db: Session = Depends(get_db)):
+    try:
+        event = get_event_by_id(db, event_id)
+    except NoResultFound:
+        raise HTTPException(status_code=404, detail="Event not found")
+    except MultipleResultsFound:
```

Comment:
There can be no multiplication id in db

Reply:
True, I used the exception in case something went terribly wrong and somehow an event id did duplicate. When using .one() this will throw this exception. Using .first() would have return an arbitrary one, which would be more dangerous as it would potentially show the wrong event to the user

Reply:
This is not a possible situation on the part of db

Reply:
Maybe because there is nothing general that will catch, it is right to leave the catch